axon-server-connector/src/main/java/org/axonframework/axonserver/connector/processor/EventProcessorControlService.java

```diff
@@ -157,9 +173,13 @@ private void mergeSegment(PlatformOutboundInstruction platformOutboundInstructio
         String processorName = mergeSegment.getProcessorName();
         int segmentId = mergeSegment.getSegmentIdentifier();
         try {
-            eventProcessorController.mergeSegment(processorName, segmentId);
+            if (!eventProcessorController.mergeSegment(processorName, segmentId)) {
```

Comment:
The `EventProcessorController#mergeSegment(String, int)` function will also return `false` if the underlying `TrackingEventProcessor` couldn't fulfill the request. Thus stating it's an `IllegalArgumentException` because the given `processorName` is not a `TrackingEventProcessor` is not always correct.